src/core/sig_state.ml

```diff
@@ -7,112 +7,180 @@ open Files
 open Pos
 open Syntax
 open Terms
+open Sign
 
 (** Pretty-printing information associated to a symbol. *)
 type pp_hint =
-  | No_hint
+  | Unqual
   | Prefix of unop
   | Infix of binop
+  | Zero
+  | Succ
+
+(** [eq_pp_hint h1 h2] says whether [h1] and [h2] are equal, ignoring
+   associativity and priorities. *)
+let eq_pp_hint : pp_hint eq = fun h1 h2 ->
+  match (h1, h2) with
+  | (Unqual, Unqual)
+  | (Zero, Zero)
+  | (Succ, Succ) -> true
+  | (Prefix (s1,_,_), Prefix (s2,_,_))
+  | (Infix (s1,_,_,_), Infix (s2,_,_,_)) -> s1 = s2
+  | (_, _) -> false
+
+type in_scope_map = (sym * Pos.popt) StrMap.t
+type pp_hint_map = pp_hint SymMap.t
+type path_map = string PathMap.t
 
 (** State of the signature, including aliasing and accessible symbols. *)
 type sig_state =
-  { signature : Sign.t                    (** Current signature.   *)
-  ; in_scope  : (sym * Pos.popt) StrMap.t (** Symbols in scope.    *)
-  ; aliases   : Path.t StrMap.t           (** Established aliases. *)
-  ; builtins  : Sign.builtin_map          (** Builtin symbols.     *)
-  ; pp_hints  : pp_hint SymMap.t          (** Printing hints.      *) }
+  { signature : Sign.t          (** Current signature.   *)
+  ; in_scope  : in_scope_map    (** Symbols in scope.    *)
+  ; aliases   : Path.t StrMap.t (** Established aliases. *)
+  ; path_map  : path_map        (** Reverse map of [aliases]. *)
+  ; builtins  : builtin_map     (** Builtin symbols.     *)
+  ; unops     : sym StrMap.t    (** Unary operators.     *)
+  ; binops    : sym StrMap.t    (** Binary operators.    *)
+  ; pp_hints  : pp_hint_map     (** Printing hints.      *) }
```

Comment:
Closing comments may be aligned using a rectangular selection (`C-x SPC` or `M-x rectangle-mark-mode`) then `M-x align-regexp RET \*)`.

Reply:
Arrows `->` can be aligned as well with this.

Reply:
Ok. But all these alignment constraints artificially increase diff's...

Reply:
True, but github allows to hide whitespace change (although you have to reset the option each time...)